test/SILOptimizer/cast_foldings.sil

```diff
@@ -145,7 +145,7 @@ bb0(%0 : $*AnyObject):
 // CHECK:  store [[TMP]] to %0 : $*Optional<AnyObject>
 sil @testIOU : $@convention(thin) (@in Optional<AnyObject>) -> (@out Optional<AnyObject>) {
 bb0(%0 : $*Optional<AnyObject>, %1: $*Optional<AnyObject>):
-  unconditional_checked_cast_addr ImplicitlyUnwrappedOptional<AnyObject> in %1 : $*Optional<AnyObject> to Optional<AnyObject> in %0 : $*Optional<AnyObject>
+  unconditional_checked_cast_addr Swift.ImplicitlyUnwrappedOptional<AnyObject> in %1 : $*Optional<AnyObject> to Optional<AnyObject> in %0 : $*Optional<AnyObject>
```

Comment:
@rudkx This was a hack to keep this test doing what it's doing, but it's not *supposed* to work, is it? What if someone's doing this in regular Swift code?

Reply:
Ah, it looks like this busts the parse_stdlib tests too. I'm going to remove the restriction on mentioning IUO by name in SIL mode instead, but you still might want to look at this for non-SIL mode.